src/components/Personalization/index.js

```diff
@@ -10,14 +10,31 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { isNonEmptyArray } from "../../utils";
+import { isNonEmptyArray, uuid, getTopLevelCookieDomain } from "../../utils";
 import { initRuleComponentModules, executeRules } from "./turbine";
 import { hideContainers, showContainers, hideElements } from "./flicker";
 
 const PAGE_HANDLE = "personalization:page";
+const SESSION_ID_COOKIE = "alloy-session-id";
+const SESSION_ID_TTL = 31 * 60 * 1000;
 const EVENT_COMMAND = "event";
 const isElementExists = event => event.moduleType === "elementExists";
 
+const getOrCreateSessionId = cookie => {
+  let sessionId = cookie.get(SESSION_ID_COOKIE);
+  const domain = getTopLevelCookieDomain(window, cookie);
+  const expires = new Date(Date.now() + SESSION_ID_TTL);
+
+  if (!sessionId) {
+    sessionId = uuid();
+  }
+
+  // We have to extend cookie lifetime
+  cookie.set(SESSION_ID_COOKIE, sessionId, { domain, expires });
```

Comment:
I believe `cookie` is the alloy cookie accessor object (found in `src/core/createCookie.js`), which doesn't accept a `domain` or `expires`. Since the alloy cookie is a plethora of data, it all shares the same expiry (and domain), which is `180` days.

Reply:
I have removed the domain and expires and I have added expires as part of the cookie value.